src/librbd/internal.cc

```diff
@@ -2838,6 +2850,51 @@ int mirror_image_disable_internal(ImageCtx *ictx, bool force,
       return -EINVAL;
     }
 
+    // is mirroring  enabled for the child?
+    vector < librbd::snap_info_t > snaps;
+    r = snap_list(ictx, snaps);
+    if (r < 0)
+      return r;
+
+    for (auto &snap : snaps) {
+      set < pair < string,string > >  names;
+      ictx->snap_id = snap.id;
```

Comment:
Not necessarily safe to change the active snap id -- it would be an unexpected consequence.